### PR TITLE
feat: refactoring action to use package

### PR DESCRIPTION
**Pull request by dciborow**

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -105,36 +105,31 @@ inputs:
 runs:
   using: "composite"
   steps:
-      - name: Set up Python 3.8
+      - name: Set up Python 3.11
         uses: actions/setup-python@v2
         with:
-          python-version: 3.8
+          python-version: 3.11
+          cache: pip
```
Comment (dciborow) [inline action.yml:112]:
```suggestion
```

File: action.yml [hunk 2]
```diff
@@ -105,36 +105,30 @@ inputs:
...
         shell: bash
         run: |
           source .env/bin/activate
-          
-          ORG_NAME=$(echo ${BRANCH%/*})
-          curl --request GET --url https://raw.githubusercontent.com/$ORG_NAME/gpt-review/$ACTION_REF/review.py > review.py
 
-          python review.py
+          pip install -e .
+
```
Comment (dciborow) [inline action.yml:127]:
```suggestion
```

Comment (dciborow) [inline action.yml:127]:
```suggestion
```

File: tests/test_review.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+@pytest.mark.integration
+def test_int_get_review(force_gpt_3: None) -> None:
+    get_review_test()
+
+
+def get_review_test() -> None:
+    """Test get_review."""
+    _GitHubClient.post_pr_summary(
+        """
+diff --git a/README.md b/README.md
```
Comment (dciborow) [inline tests/test_review.py:19]: extract to file

File: tests/conftest.py [hunk 4]
```diff
@@ -41,3 +41,56 @@ def from_documents(documents, service_context=None) -> MockIndex:
...
+
+    def mock_init(cls):
+        return MockGit()
+
+    monkeypatch.setattr("git.repo.Repo.init", mock_init)
+
+
+@pytest.fixture
+def force_gpt_3(monkeypatch) -> None:
+    monkeypatch.setenv("GPT3_ONLY", "1")
```
Comment (dciborow) [inline tests/conftest.py:96]:
```suggestion
```

**Review by danay1999 — APPROVED** [review]
Looks good to me

**Review by deepika087 — APPROVED** [review]

File: tests/conftest.py [hunk 5]
```diff
@@ -44,3 +44,51 @@ def from_documents(documents, service_context=None) -> MockIndex:
...
+        def commit(self, message) -> str:
+            return "test commit response"
+
+        def diff(self, message, cached) -> str:
+            return "test diff response"
+
+    def mock_init(cls):
+        return MockGit()
+
+    monkeypatch.setattr("git.repo.Repo.init", mock_init)
```
[the remark this answers is not in the document]
Comment (dciborow) [reply tests/conftest.py:94]:
```suggestion
    monkeypatch.setattr("git.repo.Repo.init", mock_init)

```

**Pull request merged**